### PR TITLE
feat: add filtering to ce runtime list workflow runs

### PR DESCRIPTION
**Pull request by BenjaminMummery**
# The problem

The public API for listing workflows has some filtering options (max_age, limit, state). CERuntime.list_workflow_runs() does not make use of the options.

# This PR's solution

Modifies the workflow listing code to include these parameters in the API request.

# Checklist

_Check that this PR satisfies the following items:_

- [X] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [X] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [X] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

[ORQSDK-684]

[ORQSDK-684]: https://zapatacomputing.atlassian.net/browse/ORQSDK-684?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/orquestra/sdk/_base/_driver/_client.py [hunk 1]
```diff
@@ -658,6 +676,18 @@ def _list_workflow_runs(
...
+                    ",".join(
+                        (
+                            [state.value]
+                            if not isinstance(state, list)
+                            else [st.value for st in state]
+                        )
+                    )
+                    if state
+                    else None
+                ),
```
Comment (jamesclark-Zapata) [inline src/orquestra/sdk/_base/_driver/_client.py:690]: **suggestion:** ‏This seems just enough complexity to pull out.

```
def _get_state_query(state: Optional[Union[State, List[State]]]):
    if state is None:
        return None
    if isinstance(state, State):
        return state.value
    else:
        return ",".join(st.value for st in state)
```

then call with:
```suggestion
                state=_get_state_query(state),
```

File: src/orquestra/sdk/_base/_driver/_models.py [hunk 2]
```diff
@@ -309,14 +309,26 @@ class CreateWorkflowRunResponse(pydantic.BaseModel):
 class ListWorkflowRunsRequest(pydantic.BaseModel):
     """
     Implements:
-    https://github.com/zapatacomputing/workflow-driver/blob/34eba4253b56266772795a8a59d6ec7edf88c65a/openapi/src/resources/workflow-runs.yaml#L9.
+    https://github.com/zapatacomputing/workflow-driver/blob/c52013c0f4df066159fc32ad38d489b3eaff5850/openapi/src/resources/workflow-runs.yaml#L14.
+
+    Returns all workflow runs along with their definition IDs, statuses and owners.
```
Comment (jamesclark-Zapata) [inline src/orquestra/sdk/_base/_driver/_models.py:314]: **nitpick:** ‏Technically this doesn't return anything.
If you want to reference the what is returned, you could add a "See also" but saying this "returns" is misleading.

File: tests/sdk/driver/test_ce_runtime.py [hunk 3]
```diff
@@ -990,58 +990,101 @@ def test_happy_path(
             page_size=None,
             page_token=None,
             workspace=None,
+            max_age=None,
+            state=None,
         )
         assert runs == wf_runs
 
     @pytest.mark.parametrize(
```
Comment (jamesclark-Zapata) [inline tests/sdk/driver/test_ce_runtime.py:998]: **thought:** ‏These looks like two very similar tests. Maybe an area for improvement another time.

File: src/orquestra/sdk/_base/_driver/_ce_runtime.py [hunk 4]
```diff
@@ -465,12 +460,13 @@ def _list_wf_runs(
 
         for page_size in page_sizes:
             try:
-                # TODO(ORQSDK-684): driver client cannot do filtering via API yet
-                # https://zapatacomputing.atlassian.net/browse/ORQSDK-684?atlOrigin=eyJpIjoiYmNiZjUyMjZiNzg5NDI2YWJmNGU5NzAxZDI1MmJlNzEiLCJwIjoiaiJ9 # noqa: E501
+                # https://github.com/zapatacomputing/workflow-driver/blob/c52013c0f4df066159fc32ad38d489b3eaff5850/openapi/src/resources/workflow-runs.yaml#L14 # noqa: E501
```
Comment (jamesclark-Zapata) [inline src/orquestra/sdk/_base/_driver/_ce_runtime.py:463]: **nitpick:** ‏Usually we document these params closer to the API boundary, such as the `_models` module.

**Review by jamesclark-Zapata — CHANGES_REQUESTED** [review]
Do we need any new tests for `_client`?

File: src/orquestra/sdk/_base/_driver/_client.py (hunk 1 above)
Comment (BenjaminMummery) [reply src/orquestra/sdk/_base/_driver/_client.py]: [5e87105](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/335/commits/5e871059eb337be15fae6ef2e0055197a2a651ef)

File: src/orquestra/sdk/_base/_driver/_models.py (hunk 2 above)
Comment (BenjaminMummery) [reply src/orquestra/sdk/_base/_driver/_models.py]: [e9ae973](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/335/commits/e9ae97358d6fdc2c2597aaedbeab7e5321e83687)

File: src/orquestra/sdk/_base/_driver/_ce_runtime.py (hunk 4 above)
Comment (BenjaminMummery) [reply src/orquestra/sdk/_base/_driver/_ce_runtime.py]: [497750f](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/335/commits/497750f8df026839566c30b80c4611f5d447e936)

**Comment by BenjaminMummery** [comment]
> Do we need any new tests for `_client`?

Good idea, added some test cases in [7fa0510](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/335/commits/7fa05101580a2d46346e2f4fd00139ca83502523)

**Review by jamesclark-Zapata — APPROVED** [review]

**Pull request merged**